### PR TITLE
Remove goog.global

### PR DESCRIPTION
**Pull request by nicholas-l**
Not sure on the best way to do this, thoughts?

### CODE REVIEW

**Comment by nicholas-l** [comment]
I imagine there will be a bit of discussion about this as a lot of code relies on the ability to create document elements etc.

File: externs/olx.js [hunk 1]
```diff
@@ -7441,3 +7441,9 @@ olx.style.AtlasManagerOptions.prototype.maxSize;
  * @api
  */
 olx.style.AtlasManagerOptions.prototype.space;
+
+
+/**
+ * @type {Object}
+ */
+var global;
```
Comment (nicholas-l) [inline externs/olx.js]: Not sure how this should be done as I got an error about ```global``` being undefined.

**Comment by ahocevar** [comment]
I think this is the right approach. I'm only surprised about the change to `externs/olx.js`. Was that just to make the compiler happy?

**Comment by nicholas-l** [comment]
Yea otherwise it errors out:
```
info ol Compiling 315 sources
ERR! compile /home/nicholas/Development/ol3/src/ol/ol.js:278: ERROR - variable global is undeclared
ERR! compile   if (typeof global !== 'undefined') {
ERR! compile              ^
ERR! compile 
ERR! compile 
ERR! compile 1 error(s), 
ERR! compile 0 warning(s)
ERR! compile 
ERR! Process exited with non-zero status, see log for more detail: 1 ```

**Comment by ahocevar** [comment]
I think this is a bug in the closure compiler. It should know about `global`. If I am correct, then `global` should be defined in `externs/closure-compiler.js`, not `externs/olx.js`. cc @fredj - what does our Closure expert say?

**Comment by fredj** [comment]
This can be simplified to `ol.global = this;`
If we are in a nodejs environment, `this` will be `global` and `window` if we are in a browser.

That's how it's done in closure-library

**Comment by nicholas-l** [comment]
Initially I thought that ```ol.global = this``` this but in a module environment this is meant to be ```undefined``` according to http://exploringjs.com/es6/ch_modules.html#sec_modules-vs-scripts

**Comment by ahocevar** [comment]
@fredj If you search for 'global' in the compiled code, you'll see that the compiler does something along the lines of what is done in this pull request.

**Comment by marcjansen** [comment]
would

```js
ol.global = (function() {
  return this;
}());
```

work everywhere?

**Comment by fredj** [comment]
In es5 strict mode, `this` is also `undefined`; I'm wondering how the compiler handles this

**Comment by marcjansen** [comment]
Obscure, but

```
ol.global = Function('return this')()
```

looks OK for me everyhwere. (Idea from http://stackoverflow.com/a/3277192)

File: externs/closure-compiler.js [hunk 2]
```diff
@@ -15,3 +15,9 @@
  * @see http://www.w3.org/TR/pointerevents/#the-touch-action-css-property
  */
 CSSProperties.prototype.touchAction;
+
+
+/**
+ * @type {Object}
+ */
+var global;
```
Comment (ahocevar) [inline externs/closure-compiler.js]: This should no longer be needed now.

**Comment by ahocevar** [comment]
I think this looks great now. Thanks @nicholas-l, and thanks @marcjansen for the research.

**Pull request merged**